apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/jvm/JVMUtil.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.agent.core.jvm;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import java.lang.management.ManagementFactory;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import org.apache.skywalking.apm.agent.core.logging.api.ILog;
+import org.apache.skywalking.apm.agent.core.logging.api.LogManager;
+import org.apache.skywalking.apm.agent.core.util.CollectionUtil;
+import org.apache.skywalking.apm.network.common.v3.KeyStringValuePair;
+
+public class JVMUtil {
+
+    private static final ILog LOGGER = LogManager.getLogger(JVMUtil.class);
+    private final static String PATH_SEPARATOR = "/";
+    private final static String JAR_SEPARATOR = "!";
+    private static List<String> LAST_LIB_JAR_NAMES = new ArrayList<>();
+    private static Set<ClassLoader> CURRENT_CLASSLOADER_LIST = new HashSet<>();
+
+    public static Set<ClassLoader> getCurrentClassloaderList() {
+        return CURRENT_CLASSLOADER_LIST;
+    }
+
+    /**
+     * Build the required JVM information to add to the instance properties
+     */
+    public static List<KeyStringValuePair> buildJVMInfo() {
+        List<KeyStringValuePair> jvmInfo = new ArrayList<>();
+        jvmInfo.add(KeyStringValuePair.newBuilder().setKey("Start Time").setValue(getVmStartTime()).build());
+        Gson gson = new GsonBuilder().disableHtmlEscaping().create();
+        jvmInfo.add(KeyStringValuePair.newBuilder().setKey("JVM Arguments").setValue(gson.toJson(getVmArgs())).build());
+        List<String> libJarNames = getLibJarNames();
```

Comment:
You are doing this every time, don't ever consider performance impact. This mechanism would be very dangerous in the prod env.